### PR TITLE
Receipts storage optimization

### PR DESCRIPTION
**Pull request by LukaszRozmej**

### CODE REVIEW

File: src/Nethermind/Nethermind.Runner/Ethereum/Steps/ResetDatabaseMigrations.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        {
+            _receiptStorage = _context.ReceiptStorage ?? throw new StepDependencyException(nameof(_context.ReceiptStorage));
+            _blockTree = _context.BlockTree ?? throw new StepDependencyException(nameof(_context.BlockTree));
+            _chainLevelInfoRepository = _context.ChainLevelInfoRepository ?? throw new StepDependencyException(nameof(_context.ChainLevelInfoRepository));
+            
+            var initConfig = _context.Config<IInitConfig>();
+
+            if (initConfig.StoreReceipts)
+            {
+                ResetMigrationIndexIfNeeded();
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/Ethereum/Steps/ResetDatabaseMigrations.cs:53]: where is the actual migration logic here?

File: src/Nethermind/Nethermind.Serialization.Rlp/ReceiptStorageDecoder.cs [hunk 2]
```diff
@@ -23,6 +23,14 @@ namespace Nethermind.Serialization.Rlp
 {
     public class ReceiptStorageDecoder : IRlpDecoder<TxReceipt>
     {
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Serialization.Rlp/ReceiptStorageDecoder.cs:25]: Is eth63 GetReceipts served correctly - not sure if this is properly tested

File: src/Nethermind/Nethermind.Runner/Ethereum/Steps/ResetDatabaseMigrations.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Runner/Ethereum/Steps/ResetDatabaseMigrations.cs:53]: Migration logic is in DatabaSeMigrations step.

File: src/Nethermind/Nethermind.Serialization.Rlp/ReceiptStorageDecoder.cs (hunk 2 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Serialization.Rlp/ReceiptStorageDecoder.cs:25]: This is special decoder only for storage. There is separate ReceiptMessageDecoder for network communications

File: src/Nethermind/Nethermind.Blockchain/BlockTree.cs [hunk 3]
```diff
@@ -1259,18 +1238,12 @@ public Block FindBlock(Keccak blockHash, BlockTreeLookupOptions options)
                 return null;
             }
 
-            Block block = _blockCache.Get(blockHash);
+            Block block = _blockDb.Get(blockHash, _blockDecoder, _blockCache, false);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Blockchain/BlockTree.cs:1241]: just important thing to remember is not to cache headers that do not have total diffuclty set

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Blockchain/BlockTree.cs:1241]: yep there is explicit false not to cache at this call

**Pull request merged**